packages/bus/src/Brokers/RMQEventBus.ts

```diff
@@ -0,0 +1,395 @@
+    return Promise.race([
+      once(this.subject, correlationId) as Promise<[R]>,
+      new Promise<never>((_, reject) =>
+        setTimeout(reject, ttl, new NoResponse(ttl))
+      )
+    ]);
+  }
+
+  private reflectEventsNames<T, R>(
+    handler: EventHandlerConstructor<T, R>
+  ): string[] {
+    const types: (new (...args: unknown[]) => Event<T> | string)[] =
+      Reflect.getMetadata(Event, handler) ?? [];
+
+    return types.map((event: new (...args: unknown[]) => Event<T> | string) =>
+      typeof event === 'string' ? event : event.name
+    );
+  }
+
+  private async createConsumerChannel(): Promise<void> {
+    if (this.channel) return;
+
+    if (!this.client) {
+      throw new Error('The client was not created.');
+    }
+
+    this.channel = await this.client.createConfirmChannel();
+
+    await this.bindExchangesToQueue();
+    await this.startBasicConsume();
+    await this.startReplyQueueConsume();
+  }
+
+  private async startReplyQueueConsume(): Promise<void> {
+    if (!this.channel) {
+      return;
+    }
+
+    const { consumerTag } = await this.channel.consume(
+      this.REPLY_QUEUE_NAME,
+      (msg: ConsumeMessage | null) => this.processReply(msg),
+      {
+        noAck: true
+      }
+    );
+
+    this.consumerTags.push(consumerTag);
+  }
+
+  private async startBasicConsume(): Promise<void> {
+    if (!this.channel) {
+      return;
+    }
+
+    const { consumerTag } = await this.channel.consume(
+      this.configuration.clientQueue,
+      (msg: ConsumeMessage | null) => this.consumeReceived(msg),
+      {
+        noAck: true
+      }
+    );
+
+    this.consumerTags.push(consumerTag);
+  }
+
+  private async bindExchangesToQueue(): Promise<void> {
+    await this.channel?.assertExchange(this.configuration.exchange, 'direct', {
+      durable: true
+    });
+
+    await this.channel?.assertQueue(this.configuration.clientQueue, {
+      durable: true,
+      exclusive: false,
+      autoDelete: true
+    });
+  }
+
+  private processReply(message: ConsumeMessage | null): void {
+    const event: ParsedConsumeMessage | undefined =
+      this.parseConsumeMessage(message);
+
+    if (event && event.correlationId) {
+      this.subject.emit(event.correlationId, event.payload);
+    }
+  }
+
+  private async consumeReceived(message: ConsumeMessage | null): Promise<void> {
+    const event: ParsedConsumeMessage | undefined =
+      this.parseConsumeMessage(message);
+
+    if (!event) {
+      return;
+    }
+
+    const handlers = this.handlers.get(event.name);
+
+    if (!handlers) {
+      return;
+    }
+
+    for (const handler of handlers) {
+      await this.provideHandler(handler, event);
+    }
+  }
+
+  private async provideHandler(
+    handler: EventHandler<unknown, unknown>,
+    event: ParsedConsumeMessage
+  ): Promise<void> {
+    try {
+      const response = await handler.handle(event.payload);
+
+      if (!event.replyTo) {
+        return;
+      }
```

Comment:
`response` may be null or undefined

```suggestion
      if (!response || !event.replyTo) {
        return;
      }
```